examples/download_progress/src/download.rs

```diff
@@ -1,89 +1,105 @@
-use iced_futures::futures;
+use std::hash::Hash;
 
-// Just a little utility function
-pub fn file<T: ToString>(url: T) -> iced::Subscription<Progress> {
-    iced::Subscription::from_recipe(Download {
-        url: url.to_string(),
-    })
-}
```

Comment:
I believe exposing a function to create a `Subscription` is a simpler API than exposing the `Download` internals.